File: MarketFlutter/lib/models/billing_details.dart

class BillingDetails {
   int id;
   String fullName;
   String address;
   String city;
   String postalCode;
   String phoneNumber;
   String email;
   String userId;

  BillingDetails({
    required this.id,
    required this.fullName,
    required this.address,
    required this.city,
    required this.postalCode,
    required this.phoneNumber,
    required this.email,
    required this.userId,
  });

  // Factory constructor for creating a BillingDetails object from JSON
  factory BillingDetails.fromJson(Map<String, dynamic> json) {
    return BillingDetails(
      id: json['id'],
      fullName: json['fullName'],
      address: json['address'],
      city: json['city'],
      postalCode: json['postalCode'],
      phoneNumber: json['phoneNumber'],
      email: json['email'],
      userId: json['userId'],
    );
  }

  // Method to convert a BillingDetails object to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'fullName': fullName,
      'address': address,
      'city': city,
      'postalCode': postalCode,
      'phoneNumber': phoneNumber,
      'email': email,
      'userId': userId,
    };
  }
}

File: MarketPortal/Market/Data/Models/BillingDetails.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Market.Data.Models;

namespace Market.Data.Models
{
    public class BillingDetails
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(100)]
        public string FullName { get; set; } = string.Empty;

        [Required]
        [MaxLength(200)]
        public string Address { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string City { get; set; } = string.Empty;

        [Required]
        [MaxLength(20)]
        public string PostalCode { get; set; } = string.Empty;

        [Required]
        [MaxLength(15)]
        public string PhoneNumber { get; set; } = string.Empty;

        [Required]
        [MaxLength(50)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [ForeignKey(nameof(User))]
        public Guid UserId { get; set; }
        public virtual User? User { get; set; }

        public virtual List<Order>? Orders { get; set; }

        public virtual List<Purchase>? Purchases { get; set; }
    }
}
